File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53403/output/project/src/customerimporter/cuit.py
from customerimporter.identification import Identification


class Cuit(Identification):

    # type

    @classmethod
    def type_code(cls):
        return "C"

    @classmethod
    def valid_headers(cls):
        return ["20", "23", "24", "25", "26", "27", "30", "33", "34"]

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid CUIT number"

    # validation

    @classmethod
    def assert_is_valid(cls, a_number):
        # "23-25666777-9" size 13
        cls.assert_size_is_between(a_number, 12, 13)
        cls.assert_that(a_number[2] == "-" and a_number[-2] == "-")
        cls.assert_that(a_number[:2] in cls.valid_headers())
        cls.assert_that(a_number[-1].isdigit())
        cls.assert_all_are_digits(a_number[3:-2])

    # testing

    def is_cuit(self):
        return True

    # number

    def cuit_number_if_none(self, a_none_closure):
        return self.number()

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53404/output/project/src/main/java/customerimporter/CUIT.java
package customerimporter;

import static customerimporter.Validation.assertThat;
import static customerimporter.Validation.isAllDigits;
import static customerimporter.Validation.isBetween;

import java.util.List;
import java.util.function.Supplier;

public class CUIT extends Identification {

    public static final String TYPE_CODE = "C";

    // "23-25666777-9" size 13
    private static final int MINIMUM_SIZE = 12;
    private static final int MAXIMUM_SIZE = 13;
    private static final List<String> validHeaders = List.of("20", "23", "24", "25", "26", "27", "30", "33", "34");

    // instance creation

    public static CUIT with(String aNumber) {
        assertIsValidNumber(aNumber);
        return new CUIT(aNumber);
    }

    // error messages

    public static String invalidNumberErrorDescription() {
        return "Invalid CUIT number";
    }

    // asserting - private

    private static void assertIsValidNumber(String aNumber) {
        assertThat(isBetween(aNumber.length(), MINIMUM_SIZE, MAXIMUM_SIZE), invalidNumberErrorDescription());
        assertThat(aNumber.charAt(2) == '-' && aNumber.charAt(aNumber.length() - 2) == '-', invalidNumberErrorDescription());
        assertThat(validHeaders.contains(aNumber.substring(0, 2)), invalidNumberErrorDescription());
        assertThat(Character.isDigit(aNumber.charAt(aNumber.length() - 1)), invalidNumberErrorDescription());
        assertThat(isAllDigits(aNumber.substring(3, aNumber.length() - 2)), invalidNumberErrorDescription());
    }

    // initialization

    private CUIT(String aNumber) {
        super(aNumber);
    }

    // accessing

    public String typeCode() {
        return TYPE_CODE;
    }

    // testing

    public boolean isCUIT() {
        return true;
    }

    public boolean isDNI() {
        return false;
    }

    // evaluating

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return number();
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }
}
